File: 课程网站/microlectures/2.2-sequences/source/first_experiment.py
from pathlib import Path
import numpy as np
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt
mode='pi';shift=20;n=np.arange(120)
omega=.3*np.pi if mode=='pi' else .3
x=np.cos(omega*n);y=np.cos(omega*(n+shift))
print(f'shift={shift} max_difference={np.max(np.abs(y-x)):.12g}')
plt.stem(n[:41],x[:41],linefmt='C0-',markerfmt='C0o',label='x[n]');plt.stem(n[:41],y[:41],linefmt='C1-',markerfmt='C1.',label='x[n+N]');plt.xlabel('n');plt.ylabel('amplitude');plt.legend();plt.grid()
plt.savefig(Path(__file__).resolve().parent/'first-python.png');plt.close()

File: 课程网站/microlectures/2.2-sequences/source/first_experiment.m
% 修改mode为pi/rad，shift为7/20/21/40。有限数值检查不替代理论证明。
mode='pi';shift=20;n=0:119;
if strcmp(mode,'pi'),omega=.3*pi;else,omega=.3;end
x=cos(omega*n);y=cos(omega*(n+shift));
fprintf('shift=%d max_difference=%.12g\n',shift,max(abs(y-x)));
figure('Visible','off');stem(n(1:41),x(1:41));hold on;stem(n(1:41),y(1:41));xlabel('n');ylabel('amplitude');legend('x[n]','x[n+N]');grid on;
exportgraphics(gcf,fullfile(fileparts(mfilename('fullpath')),'first-matlab.png'));close(gcf);
